src/Core/Utilities/DeviceTypes.cs

```diff
@@ -4,21 +4,56 @@ namespace Bit.Core.Utilities;
 
 public static class DeviceTypes
 {
-    public static IReadOnlyCollection<DeviceType> MobileTypes { get; } = new[]
-    {
+    public static IReadOnlyCollection<DeviceType> MobileTypes { get; } =
+    [
         DeviceType.Android,
         DeviceType.iOS,
-        DeviceType.AndroidAmazon,
-    };
+        DeviceType.AndroidAmazon
+    ];
 
-    public static IReadOnlyCollection<DeviceType> DesktopTypes { get; } = new[]
-    {
+    public static IReadOnlyCollection<DeviceType> DesktopTypes { get; } =
+    [
         DeviceType.LinuxDesktop,
         DeviceType.MacOsDesktop,
         DeviceType.WindowsDesktop,
         DeviceType.UWP,
         DeviceType.WindowsCLI,
         DeviceType.MacOsCLI,
-        DeviceType.LinuxCLI,
-    };
+        DeviceType.LinuxCLI
+    ];
+
+
+    public static IReadOnlyCollection<DeviceType> BrowserExtensionTypes { get; } =
+    [
+        DeviceType.ChromeExtension,
+        DeviceType.FirefoxExtension,
+        DeviceType.OperaExtension,
+        DeviceType.EdgeExtension,
+        DeviceType.VivaldiExtension,
+        DeviceType.SafariExtension
+    ];
+
+    public static IReadOnlyCollection<DeviceType> BrowserTypes { get; } =
+    [
+        DeviceType.ChromeBrowser,
+        DeviceType.FirefoxBrowser,
+        DeviceType.OperaBrowser,
+        DeviceType.EdgeBrowser,
+        DeviceType.IEBrowser,
+        DeviceType.SafariBrowser,
+        DeviceType.VivaldiBrowser,
+        DeviceType.UnknownBrowser
+    ];
+
+    private static ClientType ToClientType(DeviceType? deviceType)
+    {
+        return deviceType switch
+        {
+            not null when MobileTypes.Contains(deviceType.Value) => ClientType.Mobile,
+            not null when DesktopTypes.Contains(deviceType.Value) => ClientType.Desktop,
+            not null when BrowserExtensionTypes.Contains(deviceType.Value) => ClientType.Browser,
+            not null when BrowserTypes.Contains(deviceType.Value) => ClientType.Web,
+            _ => ClientType.All
```

Comment:
💭 Not sure about this being the best to return as a default, but I don't have a better answer besides `null` and we don't want to deal with nullability here.